### PR TITLE
Include Respond Directives Examples for Java.

### PR DESCRIPTION
**Pull request by felipefzdz**
This PR addresses the following directives for #20466 
- respond-with-directives/respondWithDefaultHeader.rst
- respond-with-directives/respondWithDefaultHeaders.rst
- respond-with-directives/respondWithHeader.rst
- respond-with-directives/respondWithHeaders.rst

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @felipefzdz,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by ktoso** [comment]
OK TO TEST

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
+package docs.http.javadsl.server.directives;
```
Comment (ktoso) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java]: Should have license header (like all files, just copy the comment from the top - thanks!)

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        testRoute(route).run(HttpRequest.GET("/el-tonno"))
+                .assertHeaderExists("X-Fish-Name", "El Tonno")
+                .assertEntity("¡Ay blippy!");
+
+        testRoute(route).run(HttpRequest.GET("/los-tonnos"))
+                .assertHeaderExists("X-Fish-Name", "El Tonno")
+                .assertEntity("¡Ay ay blippy!");
+        //#respondWithDefaultHeaders
+    }
+}
```
Comment (ktoso) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java]: we're trying to move towards formatting with 2 spaces FYI, but don't have formatter for Java in place yet... I don't mind merging as is, but we'll reformat afterwards then

**Comment by ktoso** [comment]
LGTM, very cool, thanks a lot! Esp with the blippy comments :) I think they help a lot :)

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java (hunk 1 above)
Comment (felipefzdz) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java]: Of course, I forgot it, sorry!

Comment (ktoso) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java]: no problem at all :) thanks in advance for adding those :)

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java (hunk 2 above)
Comment (felipefzdz) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java]: I'm happy if it's reformatted whenever the formatter will be ready.

**Comment by felipefzdz** [comment]
Thanks to you for offering accessible issues for new contributors!

**Comment by ktoso** [comment]
Gladly :) It's a win-win situation :)
Would you be able to add the copyright header comment? Can be as separate commit here, and then I'll merge

**Comment by felipefzdz** [comment]
I've amended the commit with the copyright (that's why akka-ci run some minutes ago) However Github has been funny the whole day not showing properly the last updates. As you can see [here](https://github.com/felipefzdz/akka/blob/ddcb54a9f8a0e28c39275d3b3d720ffee3345fe1/akka-docs/rst/java/code/docs/http/javadsl/server/directives/RespondWithDirectivesExamplesTest.java) there is a new commit with the copyright (as it was amended by me). Let's wait until Github's 'eventual consistency' makes its magic ;)

**Comment by felipefzdz** [comment]
As it wasn't refreshed yet, I've extracted the license into a new commit, to see if that forces github to realise about my changes. Not luck so far. I'll have a look during the day and I'll let you know as soon as it will be ready :)

**Comment by felipefzdz** [comment]
@ktoso PR is ready with the copyright header :)

**Comment by ktoso** [comment]
Awesome, thanks :) Will merge after validation

**Comment by ktoso** [comment]
Thanks a lot!

**Pull request merged**